src/pages/index.js

```diff
@@ -20,80 +20,222 @@ import {
-  renderCard({
-    name: formData["card-title-input"],
-    link: formData["card-link-input"],
+    // Update user info in the DOM
+    userInfo.setUserInfo({
+      name: userData.name,
+      job: userData.about,
+    });
+
+    profileAvatar.src = userData.avatar;
+
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error("Error loading app data:", err);
   });
 
-  addFormValidator.disableSubmitButton();
-});
+// Function to render individual cards
+function renderCard(cardData) {
+  const cardElement = createCard(cardData);
+  cardSection.addItem(cardElement);
+}
 
-profilePopup.setEventListeners();
-cardPopup.setEventListeners();
+function handleDeleteClick(cardInstance) {
+  confirmDeletePopup.setSubmitAction(() => {
+    const cardId = cardInstance.getId();
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        cardInstance.removeCard();
+        confirmDeletePopup.close();
+      })
+      .catch((err) => {
+        console.error("Error deleting card:", err);
+      });
+  });
 
-// popupwithimage instance
+  confirmDeletePopup.open();
+}
 
-const popupWithImage = new PopupWithImage("#js-preview-modal");
-popupWithImage.setEventListeners();
+function handleCardLike(card) {
+  const isLiked = card.isLiked;
+  const cardId = card.getId();
 
-function handleImageClick(name, link) {
-  popupWithImage.open({ name, link });
+  const likeRequest = isLiked ? api.unlikeCard(cardId) : api.likeCard(cardId);
+
+  likeRequest
+    .then((updatedCard) => {
+      card.setLikes(updatedCard.isLiked); // just the boolean
+    })
+    .catch((err) => {
+      console.error("Error updating like status:", err);
+    });
 }
 
+
+// Function to create a new card
 function createCard(cardData) {
   const card = new Card(
     cardData,
     "#card-template",
-
-    handleImageClick
+    handleImageClick,
+    handleDeleteClick,
+    handleCardLike,
+    currentUserId
   );
-  return card.getView();
+
+  const cardElement = card.getView();
+  cardElement.setAttribute("data-id", cardData._id);
+
+  return cardElement;
 }
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement);
-    },
-  },
-  ".cards__list"
+// Function to handle image clicks and open the preview
+function handleImageClick(name, link) {
+  popupWithImage.open({ name, link });
+}
+
+// Popup handling
+const profilePopup = new PopupWithForm("#profile-edit-modal", (formData) => {
+  const updatedName = formData["profile-title-input"];
+  const updatedAbout = formData["profile-subtitle-input"];
+
+  profilePopup.setLoadingText(true); // Show "Saving..."
+
+  api
+    .updateUserProfile(updatedName, updatedAbout)
+    .then((updatedUserData) => {
+      userInfo.setUserInfo({
+        name: updatedUserData.name,
+        job: updatedUserData.about,
+      });
+      profileAvatar.src = updatedUserData.avatar;
```

Comment:
`      profileAvatar.src = updatedUserData.avatar;
`

Please, delete it


You don't need to set the avatar when you change the `name` and `job`